nixos/modules/services/networking/go-camo.nix

```
@@ -0,0 +1,52 @@
+{ lib, pkgs, config, ... }:
+with lib;
+
+let
+  cfg = config.services.go-camo;
+in
+{
+  options.services.go-camo = {
+    enable = mkEnableOption "go-camo service";
+    addressPort = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Address:Port to bind to for HTTP (default: 0.0.0.0:8080)";
+      apply = v: optionalString (v != null) "--listen=${v}";
+    };
+    sslListen = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Address:Port to bind to for HTTPS/SSL/TLS";
+      apply = v: optionalString (v != null) "--ssl-listen=${v}";
+    };
+    sslKey = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = "SSL private key path";
+      apply = v: optionalString (v != null) "--ssl-key=${v}";
+    };
+    sslCert = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = "SSL certificate path";
+      apply = v: optionalString (v != null) "--ssl-cert=${v}";
+    };
+    key = mkOption {
+      type = types.str;
+      apply = v: "--key=${v}";
+      description = "HMAC key";
```

```suggestion
      description = "HMAC key to use for signing URLs";
```
We should mention how to generate one.